Swat/SwatTreeFlydown.php

```diff
@@ -140,20 +123,17 @@ private function flattenTree(
         }
     }
 
-    // }}}
-    // {{{ public function setTree()
-
     /**
-     * Sets the tree to use for display
+     * Sets the tree to use for display.
      *
-     * @param SwatTreeFlydownNode|SwatDataTreeNode $tree the tree to use for
```

Comment:
These types are runtime checked in the method. It may make sense to keep the original type.

Reply:
If we do that in the method signature (as opposed to the docblock), then the child class' method (i.e. `SwatGroupedFlydown::getTree` needs to have the same signature, even though we only accept `SwatTreeFlydownNode`s in that case.

I feel like using the abstract `SwatTreeNode` which is parent to both of the possible types makes more sense?